cli/format/format_extended.go

```diff
@@ -0,0 +1,108 @@
+package format
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+type ExtendedFormatter struct {
+	writer io.Writer
+}
+
+func init() {
+	Formatters["extended"] = &ExtendedFormatter{writer: os.Stdout}
+}
+
+func (formatter *ExtendedFormatter) SetOutput(writer io.Writer) {
+	formatter.writer = writer
+}
+
+func (formatter *ExtendedFormatter) FormatSilences(silences []types.Silence) error {
+	w := tabwriter.NewWriter(formatter.writer, 0, 0, 2, ' ', 0)
+	sort.Sort(ByEndAt(silences))
+	fmt.Fprintln(w, "ID\tMatchers\tStarts At\tEnds At\tUpdated At\tCreated By\tComment\t")
+	for _, silence := range silences {
+		line := fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%s\t%s\t",
+			silence.ID,
+			extendedFormatMatchers(silence.Matchers),
+			FormatDate(silence.StartsAt),
+			FormatDate(silence.EndsAt),
+			FormatDate(silence.UpdatedAt),
+			silence.CreatedBy,
+			silence.Comment,
+		)
+		fmt.Fprintln(w, line)
```

Comment:
This could be smooshed into a single

```
fmt.Fprintf(
	w, 
	"%s\t%s\t%s\t%s\t%s\t%s\t%s\t",
	silence.ID,
	extendedFormatMatchers(silence.Matchers),
	FormatDate(silence.StartsAt),
	FormatDate(silence.EndsAt),
	FormatDate(silence.UpdatedAt),
	silence.CreatedBy,
	silence.Comment,
)
```